### PR TITLE
Contribution: Using InChI and InChIKey to Annotate your Chemical Entities 

### PR DESCRIPTION
**Pull request by jcuadros**
Contributed recipe on using InChI and InChIKey

### CODE REVIEW

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+You can also use the following software that you install locally to process the MOL file:
+
+-   InChI core software, <https://www.inchi-trust.org/download-latest-inchi-standard-software/>.
+-   OpenBabel, <https://openbabel.org/wiki/Main_Page>,
+
+to convert a MOL file to InChI or InChIKey.
+
+Many common molecular drawing programs can also export and import MOL files. This way. they can also be used to convert a MOL file to an InChI or InChIKey. Make sure to check that isotope and stereochemistry information is preserved when importing and exporting MOL files.
+
+## Case 3: I have the SMILES string or the InChI for my chemical
```
Comment (Smund27) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:60]: Should this be 3.1?

Comment (stuchalk) [reply book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:60]: No, I think this should be 4 and the later ones incremented by 1.

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    - Know what MOL files are and where to get them (<https://en.wikipedia.org/wiki/Chemical_table_file>, <https://discover.3ds.com/sites/default/files/2020-08/biovia_ctfileformats_2020.pdf> )
+    - Understand what a SMILES string is (<https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system>, <http://opensmiles.org/>)
+- Learning outcomes: After completing this example you should understand:
+    - What makes InChI and InChIKey convenient identifier for chemical
+    - How to obtain the InChI and the InChIKey of a chemical
+    - How to identify a chemical behind an InChI or an InChIKey
+- Citation: 'Using InChI and InChIKey to annotate your chemical entities', The IUPAC FAIR Chemistry Cookbook, https://iupac.github.io/WFChemCookbook/recipes/rec_using_inchi.html
+- Reuse: This notebook is made available under a [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/) license.
+```
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:20]: General comment that the text of each section should be on multiple lines to make it easy to read when doing a PR.

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+## Scenario
+
+So far, we are quite used to identifying chemical substances with a name, a drawing or a CAS Registry Number (CAS-RN). However, these identifiers have some issues that affect their usability for indexing and searching. A way to increase the FAIRness of our documents is to annotate our chemicals with InChI and InChiKeys, either in papers, data, or metadata files.
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:24]: I would replace chemicals with chemical substances

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+Let's do it together!
+
+## Case 1: I can draw my chemical 
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:28]: Add ' substance accurately' to the Case 1 header

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+Many molecular drawing packages like ChemDraw, <https://revvitysignals.com/products/research/chemdraw>, ChemSketch, <https://www.acdlabs.com/products/chemsketch/>, or MarvinSketch, <https://chemaxon.com/marvin>, can output the InChI and the InChIKey for a molecule drawn on them.
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:30]: Could you add this link to a list of drawing packages? https://en.wikipedia.org/wiki/Molecule_editor

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+![](../images/rec_using_inchi_fig1.png)
+
+Alternatively, we can use the CACTUS Chemical Identifier Resolver, <https://cactus.nci.nih.gov/chemical/structure> to get the InChI or the InChIKey of a drawn chemical.
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:34]: If you can find a link to a YouTube video on how to get the InChI/Key that would be good to add...

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+![](../images/rec_using_inchi_fig3.png)
+
+If you can name the compound according to IUPAC systematic nomenclature, you can use the Open Parser for Systematic IUPAC nomenclature (OPSIN), to calculate (this is not a search) the InChI and the InChiKey. OPSIN can be downloaded (accessed on the command line) or used online at <https://opsin.ch.cam.ac.uk/index.html>.
+
+![](../images/rec_using_inchi_fig4.png)
+
+## Case 3: I have a MOL file for my chemical
+
+If you have a MOL or SDF file for our molecule, you can use the UniChem service, <https://www.ebi.ac.uk/unichem/> to upload the file and obtain its InChI and InChIKey.
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:49]: Add links to both 'MOL' and 'SDF' that describe these files.

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+-   InChI core software, <https://www.inchi-trust.org/download-latest-inchi-standard-software/>.
+-   OpenBabel, <https://openbabel.org/wiki/Main_Page>,
+
+to convert a MOL file to InChI or InChIKey.
+
+Many common molecular drawing programs can also export and import MOL files. This way. they can also be used to convert a MOL file to an InChI or InChIKey. Make sure to check that isotope and stereochemistry information is preserved when importing and exporting MOL files.
+
+## Case 3: I have the SMILES string or the InChI for my chemical
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:61]: I would link 'SMILES' to a resource that describes the format.

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+![](../images/rec_using_inchi_fig6.png)
+
+Please be aware that complex InChIs can not be interpreted by the CACTUS Chemical Identifier Resolver.
+
+Again, import and export features of the common molecular drawing packages may be useful in this case. Last but not least, databases can also help find an InChI or an InChIKey from a SMILES; provided the compound is included in the database. Remember to check the responses before taking them as correct (specially for stereochemistry and isotope distribution).
+
+## Case 4: I only have the CAS-RN for my chemical
+
+If you only have the CAS-RN for your chemical, you have to resort to a database to potentially find the InChI and InChIKey. Curated chemical databases, such as the ACS Common Chemistry service, <https://commonchemistry.cas.org/> (for common chemicals) or SciFinder, <https://scifinder.cas.org>, will be among the first options to try. Other databases like PubChem, <https://pubchem.ncbi.nlm.nih.gov>, ChemSpider, <https://chemspider.com>, Wikidata, <https://www.wikidata.org/>, or Wikipedia, <https://www.wikipedia.org/>, (quote your CAS-RN) may also work in many cases for common chemicals.
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:74]: Replace CAS-RN in title and text with 'CAS Registry Number'.  Link to the CAS website.

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+![](../images/rec_using_inchi_fig7.png)
+
+## Last step: Let's check our InChI and InChIKey
+
+Once you have found the InChI and the InChIKey of a chemical species, it may be worth to double-check what we got as a result.
+
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:81]: Change 'to double-check' top 'double-checking' and 'what we' to 'what you'

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+### Quick checks
+
+Let's start with some simple checks.
+
+- Does the InChI starts with the expected characters, "InChI=1S/"?
+- Does the InChI have the expected layers, e.g., t, m , s sublayers if you are referring to a specific isomer, or i if you have an isotopically -labelled compound?
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:87]: Remove extra space after 'm'

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+- Does the InChIKey have 27 characters?
+- Does it follow the expected structure 14 uppercase letters, a hyphen, 10 uppercase letters, a hyphen and one last uppercase letter?
+- Does it end with N if the chemical specie is neutral?
+- Does the InChIKey has SA has the last two letters preceding the second dash? (SA there means version 1 standard)
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:91]: Specie should be 'species'.

File: book/contributions/rec_using_inchi/recipes/rec_using_inchi.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+### Google it!
+
+Major web search engines, like Google, Bing or Baidu can search from an InChIKey; its structure is largely specific and allows finding references and data related to your chemical specie. So, search your InChIKey and make sure you get back responses that belong to your molecule.
+
+### Use an InChI resolver
+
+Last, specific tools are being developed to resolve InChI and InChIKey by searching into databases. Currently, InChI and InChIKey resolution can be done at PubChem search page, <https://pubchem.ncbi.nlm.nih.gov/>, UniChem, <https://www.ebi.ac.uk/unichem/> or the CACTUS Chemical Identity Resolver, <https://cactus.nci.nih.gov/chemical/structure>.
+
+API to resolve programmatically InChIKeys are also being developed. Progress can be checked out at <https://inchi-resolver.org/>.
```
Comment (stuchalk) [inline book/contributions/rec_using_inchi/recipes/rec_using_inchi.md:107]: Add 'An' at the beginning of the sentence.

**Review by stuchalk — CHANGES_REQUESTED** [review]
I like the conversational style of this.  Please make the suggested changes and I will approve.

**Comment by stuchalk** [comment]
Jordi, can you make the suggested changes so I can approve and get it into the Cookbook?

**Pull request closed without merging**